src/anbox/cmds/launch.cpp

```diff
@@ -19,19 +19,36 @@
 #include "anbox/common/wait_handle.h"
 #include "anbox/dbus/stub/application_manager.h"
 #include "anbox/common/dispatcher.h"
+#include "anbox/ui/splash_screen.h"
+#include "anbox/config.h"
 #include "anbox/runtime.h"
 #include "anbox/logger.h"
 
 #include <core/dbus/asio/executor.h>
 
 #include <boost/filesystem.hpp>
 
+#include "core/posix/exec.h"
+#include "core/posix/fork.h"
 #include "core/posix/signal.h"
 
 namespace fs = boost::filesystem;
 
 namespace {
-const boost::posix_time::seconds max_wait_timeout{30};
+const boost::posix_time::seconds max_wait_timeout{240};
+const int max_restart_attempts{3};
+const std::chrono::seconds restart_interval{5};
+}
+
+bool anbox::cmds::Launch::try_launch_activity(const std::shared_ptr<dbus::stub::ApplicationManager> &stub) {
+  try {
+    DEBUG("Sending launch intent %s to Android ..", intent_);
+    stub->launch(intent_, graphics::Rect::Invalid, stack_);
+  } catch (std::exception &err) {
```

Comment:
You also might want to have a catch (...) { ... } to avoid exceptions from bubbling up.